api/app/controllers/api/v1/positions_controller.rb

```diff
@@ -28,6 +28,27 @@ def create
       end
     end
 
+    # POST /positions/import
+    def import
+      @position = Position.new(position_params)
+
+      if @position.valid?
+        if @position.save
+          render json: @position, status: :created
+        else
+          render json: @position.errors, status: :unprocessable_entity
+        end
+      else
+        @already_exists = Position.where('course_code = ? AND session_id = ?', params[:course_code], params[:session_id])
+        if @already_exists.present?
+          @already_exists.update(position_params)
+          render json: @already_exists
```

Comment:
This is a bit strange. Isn't `@already_exists` a boolean ? I would prefer a response closer to what you have above, with `render json: @position, status: :created` if possible.

Reply:
Sorry for the misleading name. The @already_exists is just a try to find the position data record according to that where condition. If it finds it, it would be a position object, if not, then just null and fails the .present call. I would try to find a better name for that.

Reply:
Ah I see, my bad I should have understood that. If a better name comes to you i'd say go for it, but its ok to leave otherwise.